h2o-r/tests/testdir_algos/coxph/runit_coxph_predict.R

```diff
@@ -2,99 +2,153 @@ setwd(normalizePath(dirname(R.utils::commandArgs(asValues=TRUE)$"f")))
 source("../../../scripts/h2o-r-test-setup.R")
 
 
+pred.r <- function(fit.8, tstdata.8) {
+    fit.8.pred <- predict(fit.8, tstdata.8, type = "lp")
+    names(fit.8.pred) <- NULL
+    return(fit.8.pred)
+}
+
+pred.h2o <- function(model, data) {
+    hex.lp <- h2o.predict(model, data)
+    as.data.frame(hex.lp)$lp
+}
+
+compare.results <- function(fit.8, hex.fit.8, tstdata.8, tstdata.hex.8) {
+    fit.8.pred <- pred.r(fit.8, tstdata.8)
+    hex.lp.8 <- pred.h2o(hex.fit.8, tstdata.hex.8)
+    expect_equal(fit.8.pred, hex.lp.8, tolerance = 1e-7, scale = 1)
+}
+
+cancer.with.sex <- function () {
+    tstdata <- cancer
+    tstdata$sex <- as.factor(tstdata$sex)
+    tstdata
+}
 
 test.CoxPH.predict <- function() {
-    pred <- function(model, data) {
-        hex.lp <- h2o.predict(model, data)
-        lp <- as.data.frame(hex.lp)$lp
-        lp[!is.na(lp)]
-    }
-
-    # Case 1: categoricals, numeric + interactions
-    tstdata.1 <- cancer
-    tstdata.1$sex <- as.factor(tstdata.1$sex)
+   
+
+    ## Case 1: categoricals, numeric + interactions
+    tstdata.1 <- cancer.with.sex()
```

Comment:
I dont think you need tstdata.N, I think you should be fine with one frame being modified for each test if necessary

Reply:
Technicaly, there is no need for it.

I would like to keep it this way anyway. Now is each test case structured in given-when-then feshion without sideeffect on data (and data is shared only when needed for comparisons - such as in the case that checks that equal weights don't afect the result).